api_tester/lib/models/purchase_configuration_api_test.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:purchases_flutter/object_wrappers.dart';
+
+// ignore_for_file: unused_element
+// ignore_for_file: unused_local_variable
+class _PurchaseConfigurationApiTest {
+  void _checkConstructor() {
+    PurchasesConfiguration configuration = PurchasesConfiguration("fakeApiKey");
+  }
+
+  void _checkProperties(PurchasesConfiguration configuration) {
+    String? appUserId = configuration.appUserID;
+    bool observerMode = configuration.observerMode;
+    String? userDefaultsSuiteName = configuration.userDefaultsSuiteName;
+    Store? store = configuration.store;
+    configuration.appUserID = null;
+    configuration.appUserID = "fakeUserId";
+    configuration.observerMode = false;
+    configuration.userDefaultsSuiteName = null;
+    configuration.userDefaultsSuiteName = "fakeSuiteName";
+    configuration.store = null;
+    configuration.store = Store.playStore;
```

Comment:
`PurchaseConfiguration` actually allows to set these properties, which I found a bit weird... We might want to mimic something like Java's `Builder` pattern in the future?

Reply:
I started doing a builder for the `PurchaseConfiguration` and then removed it since I found it clean to just use the .. notation

```
configuration = PurchasesConfiguration(StoreConfig.instance.apiKey)
         ..appUserID = null
         ..observerMode = false;
```

I hadn't thought about this case though, so I think it makes sense to go back to that builder

Reply:
Sounds good! Let me know when that's done and I can rebase my branch over your changes